app/src/main/java/com/geeksville/mesh/ui/DebugAdapter.kt

```diff
@@ -35,11 +37,63 @@ class DebugAdapter internal constructor(
     override fun onBindViewHolder(holder: DebugViewHolder, position: Int) {
         val current = logs[position]
         holder.logTypeView.text = current.message_type
-        holder.logRawMessage.text = current.raw_message
+        holder.logRawMessage.text = annotateMessage(current)
         val date = Date(current.received_date)
         holder.logDateReceivedView.text = timeFormat.format(date)
     }
 
+    /**
+     * Annotate the message with the from/to addresses in hex if they are present.
+     *
+     * This makes it easier to correlate messages with to/from in integer form and
+     * node IDs in hex form.
+     */
+    @OptIn(ExperimentalStdlibApi::class) // toHexString
+    private fun annotateMessage(current: MeshLog): CharSequence {
+        val meshPacket = current.meshPacket ?: return current.raw_message
+        val from = meshPacket.from
+        val to = meshPacket.to.let {
+            if (it == -1) UInt.MAX_VALUE else it.toUInt()
+        }
+        val fromString = "from: $from\n"
+        val fromHex = "(!${from.toHexString()})"
+        val toString = "to: $to\n"
+        val toHex = "(!${to.toHexString()})"
```

Comment:
Ah, nice!  I'll take another stab at this when I get some time.  I'm also going to try to get the work to be done upon database ingress rather than at render time, or at least see if I can lighten the load a bit.

It may take me a few days but I'll get it done. :)